x-pack/legacy/plugins/siem/public/components/events_viewer/index.tsx

```diff
@@ -131,31 +147,35 @@ const StatefulEventsViewerComponent = React.memo<Props>(
     const handleOnMouseLeave = useCallback(() => setShowInspect(false), []);
 
     return (
-      <WithSource sourceId="default">
-        {({ indexPattern, browserFields }) => (
-          <div onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave}>
-            <EventsViewer
-              browserFields={browserFields}
-              columns={columns}
-              id={id}
-              dataProviders={dataProviders!}
-              end={end}
-              filters={filters}
-              indexPattern={indexPattern}
-              isLive={isLive}
-              itemsPerPage={itemsPerPage!}
-              itemsPerPageOptions={itemsPerPageOptions!}
-              kqlMode={kqlMode}
-              onChangeItemsPerPage={onChangeItemsPerPage}
-              query={query}
-              showInspect={showInspect}
-              start={start}
-              sort={sort!}
-              toggleColumn={toggleColumn}
-            />
-          </div>
+      <div onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave}>
+        {browserFields && indexPatterns && (
+          <EventsViewer
+            browserFields={browserFields}
+            columns={
+              timelineTypeDefaultsMapping[timelineTypeContext.timelineType]?.columns ??
```

Comment:
I like the idea of using mapping here, but not quite sure about what `??` means. It is regarded as syntax error on my editor.

Reply:
@angorayc https://httptoolkit.tech/blog/5-big-features-of-typescript-3.7/

Reply:
Thanks, I followed the instruction and installed the plugin. No warnings now.
https://stackoverflow.com/questions/51741333/optional-chaining-operator-support-in-vscode